File: backend/src/main/java/com/carburant/backend/model/Notification.java
package com.carburant.backend.model;

import java.time.LocalDateTime;
import java.util.Map;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection = "notifications")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notification {
    @Id
    private String id;
    
    private String title;
    private String message;
    private NotificationType type;
    private NotificationSeverity severity;
    
    @Indexed
    private LocalDateTime timestamp;
    
    @Indexed
    private boolean read;
    
    @Indexed
    private String vehicleId;
    
    @Indexed
    private String vehicleType;
    private String region;
    private String year;
    private Map<String, Object> metadata;

    public enum NotificationType {
        ANOMALY,
        WARNING,
        INFO
    }

    public enum NotificationSeverity {
        HIGH,
        MEDIUM,
        LOW
    }
}

File: frontend/src/types/notification.ts
export type NotificationType = 'ANOMALY' | 'WARNING' | 'INFO';
export type NotificationSeverity = 'HIGH' | 'MEDIUM' | 'LOW';

export interface Notification {
    id: string;
    title: string;
    message: string;
    type: NotificationType;
    severity: NotificationSeverity;
    timestamp: string;
    read: boolean;
    vehicleId: string;
    vehicleType: string;
    region: string;
    year: string;
    metadata: Record<string, any>;
}

export interface NotificationContextType {
    notifications: Notification[];
    unreadCount: number;
    loading: boolean;
    error: string | null;
    fetchNotifications: () => Promise<void>;
    markAsRead: (id: string) => Promise<void>;
    markAllAsRead: () => Promise<void>;
    deleteNotification: (id: string) => Promise<void>;
}
